app/src/main/java/org/eyeseetea/malariacare/monitor/utils/BaseSurveyMonitor.java

```diff
@@ -0,0 +1,142 @@
+    public Survey getSurvey(){
+        return this.survey;
+    }
+
+    public boolean isRated() {
+        return (isPositive() || isNegative());
+    }
+
+    /**
+     * Tells if the given survey is positive
+     * @return
+     */
+    public boolean isPositive(){
+        return findValue(SurveyMonitor.ID_QUESTION_RDT,SurveyMonitor.ID_OPTION_RDT_POSITIVE)!=null;
+    }
+
+    /**
+     * Tells if the given survey is negative
+     * @return
+     */
+    public boolean isNegative(){
+        return findValue(SurveyMonitor.ID_QUESTION_RDT,SurveyMonitor.ID_OPTION_RDT_NEGATIVE)!=null;
+    }
+
+    /**
+     * Tells if the given survey is referral
+     * @return
+     */
+    public boolean isReferral(){
+        return findValue(SurveyMonitor.ID_QUESTION_TREATMENT,SurveyMonitor.ID_OPTION_TREATMENT_REFERRAL)!=null;
+    }
+
+    /**
+     * Tells if the given survey has Pf specie
+     * @return
+     */
+    public boolean isPf(){
+        return findValue(SurveyMonitor.ID_QUESTION_SPECIE, SurveyMonitor.ID_OPTION_SPECIE_PF)!=null;
+    }
+
+    /**
+     * Tells if the given survey has Pv specie
+     * @return
+     */
+    public boolean isPv(){
+        return findValue(SurveyMonitor.ID_QUESTION_SPECIE, SurveyMonitor.ID_OPTION_SPECIE_PV)!=null;
+    }
+
+    /**
+     * Tells if the given survey has Pf/Pv (mixed)  specie
+     * @return
+     */
+    public boolean isPfPv(){
+        return findValue(SurveyMonitor.ID_QUESTION_SPECIE, SurveyMonitor.ID_OPTION_SPECIE_PFPV)!=null;
+    }
+
+
+
+
+    /**
+     * Looks for the value with the given question + option
+     * @param idQuestion
+     * @param idOption
+     * @return
+     */
+    public Value findValue(Long idQuestion, Long idOption){
+        for(Value value:survey.getValues()){
+            if(value.matchesQuestionOption(idQuestion,idOption)){
+                return value;
+            }
+        }
+        //No matches -> null
+        return null;
+    }
+
+    /**
+     * Looks for the value with the given question  is the provided option
+     * @param idQuestion
+     * @param idOption
+     * @return
+     */
+    public boolean findOption(Long idQuestion, Long idOption){
+        Value value = findValue(idQuestion);
+        if(value==null){
+            return false;
+        }
+
+        Long valueIdOption=value.getId_option();
+        return idOption.equals(valueIdOption);
+    }
+    /**
+     * Looks for the value with the given question
+     * @param idQuestion
+     * @return
+     */
+    public Value findValue(Long idQuestion){
+        for(Value value:survey.getValues()){
+            if(value.matchesQuestion(idQuestion)){
+                return value;
+            }
+        }
+        //No matches -> null
+        return null;
+    }
+
+    /**
+     * Returns the surveys that have been sent during the last 6 months in order to create monitor stats on top of them.
+     * @return
+     */
+    public static List<Survey> findSentSurveysForMonitor() {
```

Comment:
@idelcano Don't like too much this idea of having a static method in the base class. As far as I know this method is only used from one place (it is not variant specifc). Can we extract the code from here and put it [here](https://github.com/EyeSeeTea/pictureapp/pull/351/commits/3dec5abdc7c6a26f0bf7d9b936e69f6278843c4d#diff-30d597828ea2242f465cc0c11831d0c5R99)? In general it is better to have static method in utils class